### PR TITLE
Support futures in from_delayed

### PR DESCRIPTION
**Pull request by mrocklin**
This is a bit wonky in that I had to compute before calling assert_eq.

Still this is a common tripping point.

cc @jcrist for review

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @jcrist can I get a quick yes/no on if you agree with this change?

File: dask/bag/core.py [hunk 1]
```diff
@@ -292,7 +292,9 @@ def from_delayed(value):
 
         See ``dask.bag.from_delayed`` for details
         """
-        from dask.delayed import Delayed
+        from dask.delayed import Delayed, delayed
+        if not isinstance(value, Delayed) and hasattr(value, 'key'):
+            value = delayed(value)
```
Comment (jcrist) [inline dask/bag/core.py]: Why the check for `'key'` here but not elsewhere (I assume this is to duck-type on futures)?

Comment (mrocklin) [inline dask/bag/core.py]: I've now added the check for `key` elsewhere

Comment (jcrist) [inline dask/bag/core.py]: LGTM

**Pull request merged**